Firestore/Swift/Source/PropertyWrapper/FirestoreQuery.swift

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2021 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import FirebaseFirestore
+import SwiftUI
+
+public enum FIRPredicate {
+  case isEqualTo(_ lhs: String, _ rhs: Any)
+
+  case isIn(_ lhs: String, _ rhs: [Any])
+  case isNotIn(_ lhs: String, _ rhs: [Any])
+
+  case arrayContains(_ lhs: String, _ rhs: Any)
+  case arrayContainsAny(_ lhs: String, _ rhs: [Any])
+
+  case isLessThan(_ lhs: String, _ rhs: Any)
+  case isGreaterThan(_ lhs: String, _ rhs: Any)
+
+  case isLessThanOrEqualTo(_ lhs: String, _ rhs: Any)
+  case isGreaterThanOrEqualTo(_ lhs: String, _ rhs: Any)
+
+  case orderBy(_ lhs: String, _ rhs: Bool)
+
+  case limitTo(_ lhs: Int)
+
+  /*
+   Factory methods to expose the underlying enum cases with a nicer development experience and improved semantics.
+   */
+  public static func whereField(_ lhs: String, isEqualTo rhs: Any) -> FIRPredicate {
```

Comment:
I find `rhs` and `lhs` a bit confusing. Let's something more descriptive, like `field` and `value`/`values` (see https://github.com/chFlorian/firebase-ios-sdk/pull/1/commits/41114476945ce040df34e26c6113d35b7759eef3#diff-78e468ae8bbbcd59d12162d4ba9b4399e9a4aa2c48898f52cb83bbdb7a1a6c79R37).